base/array.jl

```diff
@@ -836,9 +839,10 @@ function deleteat!(a::Vector, inds)
     return a
 end
 
+# Simpler and more efficient version for logical indexing
 function deleteat!(a::Vector, inds::AbstractVector{Bool})
     n = length(a)
-    length(inds) == n || throw(BoundsError(a, inds))
+    @boundscheck length(inds) == n || throw(BoundsError(a, inds))
     p = 1
     for (q, i) in enumerate(inds)
```

Comment:
`@boundscheck` won't have an effect here since `deleteat!` is surely too complicated to inline.